onboarding/main.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"os"
+
+	openshiftv1 "github.com/openshift/api/template/v1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/v4/api/v1"
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/v4/api/v1alpha1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	errruntime "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const (
+	onboardingTicketPublicKeySecret  = "onboarding-ticket-public-key"
+	onboardingTicketPrivateKeySecret = "onboarding-ticket-private-key"
+)
+
+var (
+	ctx                   = context.Background()
+	serverNamespace       = os.Getenv(util.WatchNamespaceEnvVar)
+	privatePem, publicPem string
+)
+
+func createSecret(cl client.Client, keyName, secretName string) {
+
+	desiredSecret := GetProviderAPIServerSecret(keyName, secretName)
+	actualSecret := &corev1.Secret{}
+
+	err := cl.Get(ctx, client.ObjectKeyFromObject(desiredSecret), actualSecret)
+
+	if err != nil && errruntime.IsNotFound(err) {
+		err = cl.Create(ctx, desiredSecret)
+		if err != nil {
+			klog.Error(err, "Failed to create secret.")
+		}
+	} else if err != nil {
+		klog.Error(err, "Failed to get secret.")
+
+	}
+}
+
+func GetProviderAPIServerSecret(key string, secretName string) *corev1.Secret {
+
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: serverNamespace,
+		},
+		Immutable: func(flag bool) *bool { return &flag }(true),
```

Comment:
We could use ["k8s.io/utils/ptr"](https://pkg.go.dev/k8s.io/utils/ptr#To)

Reply:
Sure using ptr package